### PR TITLE
Make local dev match hmcts dev workflow

### PR DESCRIPTION
**Pull request by cob16**
- add `make shell`
- add file mounting
- clean up makefile

### CODE REVIEW

File: docker/api/Dockerfile [hunk 1]
```diff
@@ -17,9 +19,9 @@ COPY . .
 
 ADD https://s3.amazonaws.com/rds-downloads/rds-combined-ca-bundle.pem ./rds-combined-ca-bundle.pem
 
-RUN chown -R 1001:appgroup /app
+RUN chown -R ${UID}:appgroup /app
 
-USER 1001
+USER UID
```
Comment (emileswarts) [inline docker/api/Dockerfile]: I forgot, didn't we need to revert this commit on the adapter?

File: Makefile [hunk 2]
```diff
@@ -1,45 +1,55 @@
...
-	$(DOCKER_COMPOSE) build --build-arg BUNDLE_FLAGS=''
+	$(DOCKER_COMPOSE) build
 
+.PHONY: serve
 serve: build
-	$(DOCKER_COMPOSE) up -d db
-	./scripts/wait_for_db.sh db postgres
-	$(DOCKER_COMPOSE) up -d app
-	$(DOCKER_COMPOSE) up -d worker
+	$(DOCKER_COMPOSE) up app
```
Comment (emileswarts) [inline Makefile]: This no longer connects to a db?  Unsure how this works.  Tests pass?

File: .circleci/config.yml [hunk 3]
```diff
@@ -4,21 +4,12 @@ jobs:
...
-        environment:
-          POSTGRES_USER: root
-          POSTGRES_DB: circle-test_test
     steps:
       - checkout
       - setup_remote_docker
       - run:
           name: test
-          command: 'make spec'
+          command: docker-compose run --rm app bundle exec rspec
```
Comment (tommotaylor) [inline .circleci/config.yml]: Why not `make spec`? Seems better to always use just one command everywhere

File: docker/api/Dockerfile (hunk 1 above)
Comment (cob16) [reply docker/api/Dockerfile]: yes but i figured out what was the problem (missing a default for circle to use)

File: .circleci/config.yml (hunk 3 above)
Comment (cob16) [reply .circleci/config.yml]: 1. `make spec` mounts the filesystem now (not supported by circle)

2. Decouples our deploys from a makefile. Devs are now free to do what they see fit in the makefile without the side effect of braking circle ci. IMO makefiles are a convenience for developers and only alias calls. having circle call them seems like a reverse dependency of this

Comment (tommotaylor) [reply .circleci/config.yml]: 👍

**Pull request merged**